src/engine/Server.h

```diff
@@ -35,6 +36,7 @@ class Server {
                cacheMaxSizeGBSingleEntry * (1ull << 30u) / sizeof(Id)),
         _allocator(ad_utility::makeAllocationMemoryLeftThreadsafeObject(
             maxMemGB * (1ull << 30u))),
+        _sortPerformanceEstimator(_allocator),
```

Comment:
A bit confusing that we `_sortPerformanceEstimator(sortPerformanceEstimator)` above and `_sortPerformanceEstimator(_allocator)` here. Not sure what to do about that though. If you think it's fine, then it's fine with me, too

Reply:
Constructing a `SortPerformanceEstimator` is expensive (it involves sorting large inputs for a time measurement). So we want to construct the Estimator (at most) once in a program.
There is only one `Server` which stores an Estimator, and passes it via (cheap) copy to the `QueryExecutionContext` class. It is however crucial, that the QueryExecutionContexts don't always recreate the estimators, as the Server creates on QueryExecutionContext per query.
This is what I got wrong initially.